File: src/models/knowledge.ts
export type VisibilityLevel = "personal" | "team" | "department" | "network";

export interface AthleticsReview {
  id: string;
  placeId: string;
  authorName: string;
  authorRole: string;
  teamName: string;
  visibility: VisibilityLevel;
  overallRating: number; // 1-5
  categoryRatings: Record<string, number>;
  body?: string;
  createdAt: string;
}

export interface PublicReview {
  id: string;
  placeId: string;
  authorName?: string;
  rating: number;
  body?: string;
}

export interface StaffNote {
  id: string;
  placeId: string;
  authorName: string;
  authorRole: string;
  teamName?: string;
  visibility: VisibilityLevel;
  body: string;
  createdAt: string;
}

export interface TeamVisit {
  id: string;
  placeId: string;
  teamName: string;
  sport: string;
  visitedOn: string; // ISO date, display as "September 2025"
}

export interface DepartmentVisitSummary {
  placeId: string;
  visitCount: number;
  lastVisitedOn?: string;
  visits: TeamVisit[];
}

export interface NetworkRecommendationSummary {
  placeId: string;
  teamsUsedCount: number;
  recommendCount: number;
  commonNotes: string[];
}

/** Composed, read-time aggregation — see docs/ARCHITECTURE.md §8. */
export interface KnowledgeSummary {
  athleticsRating?: number;
  athleticsReviewCount: number;
  departmentVisits?: DepartmentVisitSummary;
  topNotes: StaffNote[];
  networkSignal?: NetworkRecommendationSummary;
}

File: ios/SimpleScout/Models/Knowledge.swift
import Foundation

enum VisibilityLevel: String, Codable, Equatable, Hashable {
    case personal, team, department, network

    var label: String { rawValue.capitalized }
}

struct AthleticsReview: Identifiable, Codable, Equatable {
    let id: String
    let placeId: String
    let authorName: String
    let authorRole: String
    let teamName: String
    var visibility: VisibilityLevel
    let overallRating: Double
    let categoryRatings: [String: Double]
    var body: String? = nil
    let createdAt: String
}

struct StaffNote: Identifiable, Codable, Equatable {
    let id: String
    let placeId: String
    let authorName: String
    let authorRole: String
    var teamName: String? = nil
    var visibility: VisibilityLevel
    let body: String
    let createdAt: String
}

struct TeamVisit: Identifiable, Codable, Equatable {
    let id: String
    let placeId: String
    let teamName: String
    let sport: String
    let visitedOn: String // yyyy-MM-dd
}

struct DepartmentVisitSummary: Codable, Equatable {
    let placeId: String
    let visitCount: Int
    var lastVisitedOn: String? = nil
    let visits: [TeamVisit]
}

struct NetworkRecommendationSummary: Codable, Equatable {
    let placeId: String
    let teamsUsedCount: Int
    let recommendCount: Int
    let commonNotes: [String]
}

/// Composed, read-time aggregation — see docs/ARCHITECTURE.md §8.
struct KnowledgeSummary: Codable, Equatable {
    var athleticsRating: Double? = nil
    var athleticsReviewCount: Int = 0
    var departmentVisits: DepartmentVisitSummary? = nil
    var topNotes: [StaffNote] = []
    var networkSignal: NetworkRecommendationSummary? = nil
}
